pandas/core/generic.py

```diff
@@ -2324,15 +2331,44 @@ def filter(self, items=None, like=None, regex=None, axis=None):
         regex : string (regular expression)
             Keep info axis with re.search(regex, col) == True
         axis : int or None
```

Comment:
this axis args should be generated like we do for many other methods, see ``.reindex`` or ``.fillna``